core/container/externalbuilder/instance.go

```diff
@@ -108,6 +109,7 @@ func (c *ChaincodeServerUserData) ChaincodeServerInfo(cryptoDir string) (*ccintf
 		connInfo.ClientConfig.SecOpts.RequireClientCert = true
 		connInfo.ClientConfig.SecOpts.Certificate = []byte(c.ClientCert)
 		connInfo.ClientConfig.SecOpts.Key = []byte(c.ClientKey)
+		connInfo.ClientConfig.SecOpts.ServerNameOverride = c.Domain
```

Comment:
> When the chain code is deployed as an external service, the address may be an ip address rather than a domain. So we need to add a domain field so that it can pass the TLS authentication.

What exactly are you trying to solve here? How is it related to the IP address declaration and the fact you are just trying to override the server name with a different domain name? Why can you not generate a proper certificate? 

You should be able to put IP into SAN within your certificate, and if the domain is known in advance, why would you need to override it?

Reply:
Using the service name is the most convenient way to access the smart contract service in the cluster. Using registry services or K8S services, we can easily convert a service name to an IP. But resolving domain into IP addresses can be difficult.
Sometimes, we don't know the Service name when the certificate is generated. Another possibility is that we change the service name, which is more likely to happen. Generating a new certificate is not an easy task.

Reply:
Using the service name is the most convenient way to access the smart contract service in the cluster. Using registry center or K8S services, we can easily convert a service name to an IP. But resolving domain into IP addresses can be difficult.
Sometimes, we don't know the Service name when the certificate is generated. Another possibility is that we change the service name, which is more likely to happen. Generating a new certificate is not an easy task.

Reply:
Sometimes I want to use a single certificate to manage multiple smart contract services. The new smart contract service will be deployed after the certificate is generated. These new smart contract services can't use this certificate.

Reply:
Ok, so essentially, you just wanted to override the hostname while doing the TLS handshake?

Reply:
yes

Reply:
Have we considered the security implications of doing this? What is the source of `c.Domain`?

Reply:
I guess "domain" would get sourced from the connection.json as described here?
https://hyperledger-fabric.readthedocs.io/en/latest/cc_service.html

Assuming that is the case, please also update the doc in the same PR at
https://github.com/hyperledger/fabric/blob/main/docs/source/cc_service.md

Reply:
> Assuming that is the case, please also update the doc in the same PR at
yes, domain is defined in connection.json.
I'm gonna go update the doc.